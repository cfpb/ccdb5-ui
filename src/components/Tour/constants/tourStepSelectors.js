export const DOCUMENT_STEP_SELECTORS = {
  STEP_1: '.content__hero',
  STEP_2: '.m-pagination__btn-next',
  STEP_3: '.saved__search-panel div:nth-child(2)',
  STEP_4: '.back-to-results',
};

export const GEO_STEP_SELECTORS = {
  STEP_1: { selector: '.content', label: 'Index' },
  STEP_2: {
    selector: 'header.content__hero .m-list--horizontal',
    label: 'Help Links',
  },
  STEP_3: { selector: '.tabbed-navigation', label: 'Views' },
  STEP_4: { selector: '.filter-panel', label: 'Filters' },
  STEP_5: { selector: '.date-filter', label: 'Date Received' },
  STEP_6: {
    selector: '.filter-panel .aggregation.product',
    label: 'Product/Issue',
  },
  STEP_7: {
    selector: '.filter-panel .aggregation button',
    label: 'Show/Hide',
  }, // show/hide filters
  STEP_8: { selector: '.search-bar', label: 'Search Bar' },
  STEP_9: { selector: '#search-summary h2', label: 'Search Summary' },
  STEP_10: {
    selector: '.export-results button.export-btn',
    label: 'Export Data',
  },
  STEP_11: {
    selector: '.export-results button.print-preview',
    label: 'Print Page',
  }, // print button
  STEP_12: { selector: '.map-shading', label: 'Map Shading' },
  STEP_13: { selector: '#tile-chart-map', label: 'Tile Map' },
  STEP_14: { selector: '.row-chart-section', label: 'Row Charts' },
  STEP_15: { selector: '.tabbed-navigation', label: 'Tabbed Navigation' },
};

export const LIST_COMPLAINTS_SELECTORS = {
  STEP_1: { selector: '.content', label: 'Index' },
  STEP_2: {
    selector: 'header.content__hero .m-list--horizontal',
    label: 'Help Links',
  },
  STEP_3: { selector: '.tabbed-navigation', label: 'Views' },
  STEP_4: { selector: '.filter-panel', label: 'Filters' },
  STEP_5: { selector: '.date-filter', label: 'Date Received' },
  STEP_6: {
    selector: '.filter-panel .aggregation.product',
    label: 'Product/Issue',
  },
  STEP_7: {
    selector: '.filter-panel .aggregation button',
    label: 'Show/Hide',
  }, // show/hide filters
  STEP_8: { selector: '.search-bar', label: 'Search Bar' },
  STEP_9: { selector: '#search-summary h2', label: 'Search Summary' },
  STEP_10: {
    selector: '.export-results button.export-btn',
    label: 'Export Data',
  },
  STEP_11: {
    selector: '.export-results button.print-preview',
    label: 'Print Page',
  }, // print button
  STEP_12: { selector: '[data-tour="select-size"]', label: 'Show Number' },
  STEP_13: { selector: '[data-tour="select-sort"]', label: 'Sort Results' },
  STEP_14: {
    selector: '.refine-bar section.m-btn-group',
    label: 'Show Narratives',
  }, // read complaints
  STEP_15: { selector: '.cards-panel', label: 'Results' },
  STEP_16: { selector: '.to-detail', label: 'View Detailed Complaints' }, // complaint id
  STEP_17: { selector: '.tabbed-navigation', label: 'Tabbed Navigation' },
};

export const TRENDS_SELECTORS = {
  STEP_1: { selector: '.content', label: 'Index' },
  STEP_2: {
    selector: 'header.content__hero .m-list--horizontal',
    label: 'Help Links',
  },
  STEP_3: { selector: '.tabbed-navigation', label: 'Views' },
  STEP_4: { selector: '.filter-panel', label: 'Filters' },
  STEP_5: { selector: '.date-filter', label: 'Date Received' },
  STEP_6: {
    selector: '.filter-panel .aggregation.product',
    label: 'Product/Issue',
  },
  STEP_7: {
    selector: '.filter-panel .aggregation button',
    label: 'Show/Hide',
  }, // show/hide filters
  STEP_8: { selector: '.search-bar', label: 'Search Bar' },
  STEP_9: { selector: '#search-summary h2', label: 'Search Summary' },
  STEP_10: {
    selector: '.export-results button.export-btn',
    label: 'Export Data',
  },
  STEP_11: {
    selector: '.export-results button.print-preview',
    label: 'Print Page',
  }, // print button
  STEP_12: { selector: '[data-tour="select-lens"]', label: 'Aggregate By' },
  STEP_13: {
    selector: '[data-tour="select-interval"]',
    label: 'Date Interval',
  },
  STEP_14: { selector: '.row-chart-section', label: 'Row Charts' },
  STEP_15: { selector: '.tabbed-navigation', label: 'Tabbed Navigation' },
};
