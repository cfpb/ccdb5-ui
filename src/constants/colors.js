module.exports = Object.freeze({
  BriteCharts: {
    dark: '#118935',
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3',
  },
  // if you modify anything here,
  // make sure that the following colors are in sync
  // Line Chart
  // Area Chart
  // Stacked Area Chart
  // External Tooltip/Legend
  DataLens: {
    0: '#2cb34a', // 0
    1: '#addc91', // 1
    2: '#0072ce', // 4
    3: '#96c4ed', // 5
    4: '#257675', // 2
    5: '#9ec4c3', // 3
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4',
  },
});
