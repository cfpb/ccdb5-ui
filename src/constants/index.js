// Internal triggers
import { formatDate } from '../utils/formatDate';

export const DATE_RANGE_MIN = '2011-12-01';

// dayjs formatter to uses for custom validation
// https://day.js.org/docs/en/parse/string-format
export const DATE_VALIDATION_FORMAT = 'M/D/YYYY';

export const maxDate = formatDate(new Date());
export const minDate = formatDate(DATE_RANGE_MIN);

// Modal Dialogs
export const MODAL_TYPE_DATA_EXPORT = 'MODAL_TYPE_DATA_EXPORT';
export const MODAL_TYPE_EXPORT_CONFIRMATION = 'MODAL_TYPE_EXPORT_CONFIRMATION';
export const MODAL_TYPE_MORE_ABOUT = 'MODAL_TYPE_MORE_ABOUT';

// view modes
export const MODE_MAP = 'Map';
export const MODE_LIST = 'List';
export const MODE_TRENDS = 'Trends';

// map normalization modes
export const GEO_NORM_NONE = 'None';
export const GEO_NORM_PER1000 = 'Per 1000 pop.';

// Useful constants
export const NARRATIVE_SEARCH_FIELD = 'complaint_what_happened';

export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tags',
  'timely',
  'zip_code',
];

export const dateFilters = [
  'company_received_max',
  'company_received_min',
  'date_received_max',
  'date_received_min',
];

export const dateIntervals = ['Day', 'Week', 'Month', 'Quarter', 'Year'];
export const dateRanges = {
  '3m': '3 months',
  '6m': '6 months',
  '1y': '1 year',
  '3y': '3 years',
  All: 'Full date range',
};

// list of fields that we don't include in the URL or Query to keep things tidy
export const excludeFields = [
  'enablePer1000',
  'mapWarningEnabled',
  'totalPages',
  'trendsDateWarningEnabled',
];

export const flagFilters = ['has_narrative'];

// these filters we need to shim subItems when a parent is selected
export const filterPatch = ['issue', 'product'];

export const lenses = ['Overview', 'Company', 'Product'];

// note the keys need to remain strings here for the select box options
// we're declaring this so eslint autofix doesn't rename the props without
// quotes
export const sizes = {};
sizes['10'] = '10 results';
sizes['25'] = '25 results';
sizes['50'] = '50 results';
sizes['100'] = '100 results';

export const sorts = {
  created_date_desc: 'Newest to oldest',
  created_date_asc: 'Oldest to newest',
  relevance_desc: 'Relevance',
  relevance_asc: 'Relevance (asc)',
};

export const SLUG_SEPARATOR = 'â€¢';

// Keyboard
export const VK_DOWN = 40;
export const VK_ENTER = 14;
export const VK_ESCAPE = 27;
export const VK_RETURN = 13;
export const VK_TAB = 9;
export const VK_UP = 38;

/// ////////////////////////////////////////////////////////////////////////////
// JSON Objects

export const THESE_UNITED_STATES = {
  AA: 'Armed Forces Americas',
  AE: 'Armed Forces Europe',
  AL: 'Alabama',
  AK: 'Alaska',
  AS: 'American Samoa',
  AP: 'Armed Forces Pacific',
  AZ: 'Arizona',
  AR: 'Arkansas',
  CA: 'California',
  CO: 'Colorado',
  CT: 'Connecticut',
  DE: 'Delaware',
  DC: 'District Of Columbia',
  FM: 'Federated States Of Micronesia',
  FL: 'Florida',
  GA: 'Georgia',
  GU: 'Guam',
  HI: 'Hawaii',
  ID: 'Idaho',
  IL: 'Illinois',
  IN: 'Indiana',
  IA: 'Iowa',
  KS: 'Kansas',
  KY: 'Kentucky',
  LA: 'Louisiana',
  ME: 'Maine',
  MH: 'Marshall Islands',
  MD: 'Maryland',
  MA: 'Massachusetts',
  MI: 'Michigan',
  MN: 'Minnesota',
  MS: 'Mississippi',
  MO: 'Missouri',
  MT: 'Montana',
  NE: 'Nebraska',
  NV: 'Nevada',
  NH: 'New Hampshire',
  NJ: 'New Jersey',
  NM: 'New Mexico',
  NY: 'New York',
  NC: 'North Carolina',
  ND: 'North Dakota',
  MP: 'Northern Mariana Islands',
  OH: 'Ohio',
  OK: 'Oklahoma',
  OR: 'Oregon',
  PW: 'Palau',
  PA: 'Pennsylvania',
  PR: 'Puerto Rico',
  RI: 'Rhode Island',
  SC: 'South Carolina',
  SD: 'South Dakota',
  TN: 'Tennessee',
  TX: 'Texas',
  UT: 'Utah',
  VT: 'Vermont',
  VI: 'Virgin Islands',
  VA: 'Virginia',
  WA: 'Washington',
  WV: 'West Virginia',
  WI: 'Wisconsin',
  WY: 'Wyoming',
};

export const TILE_MAP_STATES = [
  'AK',
  'AL',
  'AR',
  'AZ',
  'CA',
  'CO',
  'CT',
  'DC',
  'DE',
  'FL',
  'GA',
  'HI',
  'IA',
  'ID',
  'IL',
  'IN',
  'KS',
  'KY',
  'LA',
  'MA',
  'MD',
  'ME',
  'MI',
  'MN',
  'MO',
  'MS',
  'MT',
  'NC',
  'ND',
  'NE',
  'NH',
  'NJ',
  'NM',
  'NV',
  'NY',
  'OH',
  'OK',
  'OR',
  'PA',
  'RI',
  'SC',
  'SD',
  'TN',
  'TX',
  'UT',
  'VA',
  'VT',
  'WA',
  'WI',
  'WV',
  'WY',
];

// 2017 ACS census via the census API:
// https://api.census.gov/data/2017/acs/acs5?get=B01003_001E,NAME&for=state
//
// B01003_001E is defined as TOTAL POPULATION
// (detailed here: https://api.census.gov/data/2016/acs/acs5/variables.html)
export const STATE_DATA = {
  AL: {
    abbr: 'AL',
    name: 'Alabama',
    population: '4850771',
    id: '01',
  },
  AK: {
    abbr: 'AK',
    name: 'Alaska',
    population: '738565',
    id: '02',
  },
  AZ: {
    abbr: 'AZ',
    name: 'Arizona',
    population: '6809946',
    id: '04',
  },
  AR: {
    abbr: 'AR',
    name: 'Arkansas',
    population: '2977944',
    id: '05',
  },
  CA: {
    abbr: 'CA',
    name: 'California',
    population: '38982847',
    id: '06',
  },
  CO: {
    abbr: 'CO',
    name: 'Colorado',
    population: '5436519',
    id: '08',
  },
  CT: {
    abbr: 'CT',
    name: 'Connecticut',
    population: '3594478',
    id: '09',
  },
  DE: {
    abbr: 'DE',
    name: 'Delaware',
    population: '943732',
    id: '10',
  },
  DC: {
    abbr: 'DC',
    name: 'District of Columbia',
    population: '672391',
    id: '11',
  },
  FL: {
    abbr: 'FL',
    name: 'Florida',
    population: '20278447',
    id: '12',
  },
  GA: {
    abbr: 'GA',
    name: 'Georgia',
    population: '10201635',
    id: '13',
  },
  HI: {
    abbr: 'HI',
    name: 'Hawaii',
    population: '1421658',
    id: '15',
  },
  ID: {
    abbr: 'ID',
    name: 'Idaho',
    population: '1657375',
    id: '16',
  },
  IL: {
    abbr: 'IL',
    name: 'Illinois',
    population: '12854526',
    id: '17',
  },
  IN: {
    abbr: 'IN',
    name: 'Indiana',
    population: '6614418',
    id: '18',
  },
  IA: {
    abbr: 'IA',
    name: 'Iowa',
    population: '3118102',
    id: '19',
  },
  KS: {
    abbr: 'KS',
    name: 'Kansas',
    population: '2903820',
    id: '20',
  },
  KY: {
    abbr: 'KY',
    name: 'Kentucky',
    population: '4424376',
    id: '21',
  },
  LA: {
    abbr: 'LA',
    name: 'Louisiana',
    population: '4663461',
    id: '22',
  },
  ME: {
    abbr: 'ME',
    name: 'Maine',
    population: '1330158',
    id: '23',
  },
  MD: {
    abbr: 'MD',
    name: 'Maryland',
    population: '5996079',
    id: '24',
  },
  MA: {
    abbr: 'MA',
    name: 'Massachusetts',
    population: '6789319',
    id: '25',
  },
  MI: {
    abbr: 'MI',
    name: 'Michigan',
    population: '9925568',
    id: '26',
  },
  MN: {
    abbr: 'MN',
    name: 'Minnesota',
    population: '5490726',
    id: '27',
  },
  MS: {
    abbr: 'MS',
    name: 'Mississippi',
    population: '2986220',
    id: '28',
  },
  MO: {
    abbr: 'MO',
    name: 'Missouri',
    population: '6075300',
    id: '29',
  },
  MT: {
    abbr: 'MT',
    name: 'Montana',
    population: '1029862',
    id: '30',
  },
  NE: {
    abbr: 'NE',
    name: 'Nebraska',
    population: '1893921',
    id: '31',
  },
  NV: {
    abbr: 'NV',
    name: 'Nevada',
    population: '2887725',
    id: '32',
  },
  NH: {
    abbr: 'NH',
    name: 'New Hampshire',
    population: '1331848',
    id: '33',
  },
  NJ: {
    abbr: 'NJ',
    name: 'New Jersey',
    population: '8960161',
    id: '34',
  },
  NM: {
    abbr: 'NM',
    name: 'New Mexico',
    population: '2084828',
    id: '35',
  },
  NY: {
    abbr: 'NY',
    name: 'New York',
    population: '19798228',
    id: '36',
  },
  NC: {
    abbr: 'NC',
    name: 'North Carolina',
    population: '10052564',
    id: '37',
  },
  ND: {
    abbr: 'ND',
    name: 'North Dakota',
    population: '745475',
    id: '38',
  },
  OH: {
    abbr: 'OH',
    name: 'Ohio',
    population: '11609756',
    id: '39',
  },
  OK: {
    abbr: 'OK',
    name: 'Oklahoma',
    population: '3896251',
    id: '40',
  },
  OR: {
    abbr: 'OR',
    name: 'Oregon',
    population: '4025127',
    id: '41',
  },
  PA: {
    abbr: 'PA',
    name: 'Pennsylvania',
    population: '12790505',
    id: '42',
  },
  PR: {
    abbr: 'PR',
    name: 'Puerto Rico',
    population: '3468963',
    id: '72',
  },
  RI: {
    abbr: 'RI',
    name: 'Rhode Island',
    population: '1056138',
    id: '44',
  },
  SC: {
    abbr: 'SC',
    name: 'South Carolina',
    population: '4893444',
    id: '45',
  },
  SD: {
    abbr: 'SD',
    name: 'South Dakota',
    population: '855444',
    id: '46',
  },
  TN: {
    abbr: 'TN',
    name: 'Tennessee',
    population: '6597381',
    id: '47',
  },
  TX: {
    abbr: 'TX',
    name: 'Texas',
    population: '27419612',
    id: '48',
  },
  UT: {
    abbr: 'UT',
    name: 'Utah',
    population: '2993941',
    id: '49',
  },
  VT: {
    abbr: 'VT',
    name: 'Vermont',
    population: '624636',
    id: '50',
  },
  VA: {
    abbr: 'VA',
    name: 'Virginia',
    population: '8365952',
    id: '51',
  },
  WA: {
    abbr: 'WA',
    name: 'Washington',
    population: '7169967',
    id: '53',
  },
  WV: {
    abbr: 'WV',
    name: 'West Virginia',
    population: '1836843',
    id: '54',
  },
  WI: {
    abbr: 'WI',
    name: 'Wisconsin',
    population: '5763217',
    id: '55',
  },
  WY: {
    abbr: 'WY',
    name: 'Wyoming',
    population: '583200',
    id: '56',
  },
};

export const API_PLACEHOLDER = '@@API';

// provide relative link when not local development
export const LINK_DATA_USE =
  window.location.origin.indexOf('localhost') > -1
    ? 'https://www.consumerfinance.gov/complaint/data-use/'
    : '/complaint/data-use/';
