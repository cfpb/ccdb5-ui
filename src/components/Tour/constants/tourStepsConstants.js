import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../../../constants';

import {
  GEO_STEP_SELECTORS as GEO,
  LIST_COMPLAINTS_SELECTORS as LIST,
  TRENDS_SELECTORS as TRENDS,
} from './tourStepSelectors';

const generateLinkText = (index, linkObject) =>
  `<li><button class="a-btn a-btn--link" onclick="document.querySelectorAll('.introjs-bullets li a')[${
    index + 1
  }].click()">` +
  linkObject[1].label +
  '</button></li>';

let geoIndex = '';

// we exclude the first entry so we can have Index appear as the title
Object.entries(GEO)
  .slice(1)
  .forEach((value, key) => {
    geoIndex += generateLinkText(key, value);
  });

let listIndex = '';
Object.entries(LIST)
  .slice(1)
  .forEach((value, key) => {
    listIndex += generateLinkText(key, value);
  });

let trendsIndex = '';
Object.entries(TRENDS)
  .slice(1)
  .forEach((value, key) => {
    trendsIndex += generateLinkText(key, value);
  });

export const TOUR_STEPS = {
  [MODE_MAP]: [
    {
      element: GEO.STEP_1.selector,
      intro:
        '<div>' +
        '<div class="left">' +
        'Welcome to the Consumer Complaint Database!' +
        '<br /><br />' +
        'Complaints the CFPB sends to companies for response are published in the Consumer Complaint Database after the company responds, confirming a commercial relationship with the consumer, or after 15 days, whichever comes first.' +
        '<br /><br />' +
        'Complaints the CFPB refers to other regulators, such as complaints about depository institutions with less than $10 billion in assets, are not published in the database. This database is not a statistical sample of consumers’ experiences in the marketplace.' +
        '</div>' +
        '<div class="right">' +
        '<h4>Index</h4>' +
        '<ul>' +
        geoIndex +
        '</ul></div>' +
        '</div>',
      tooltipClass: 'wide',
    },
    {
      element: GEO.STEP_2.selector,
      intro: 'These links provide more information about this database.',
    },
    {
      element: GEO.STEP_3.selector,
      intro:
        'You can change how to view complaint data by selecting the three views we have available — Trends, List, and Map.' +
        '<br /><br />' +
        'You are currently on the Map view, which displays complaints by state.' +
        '<br /><br />' +
        'To read individual complaints, select List view' +
        '<br /><br />' +
        'To visualize complaint data with product and issue breakdowns, select Trends view',
      position: 'top',
    },
    {
      element: GEO.STEP_4.selector,
      intro:
        'The “Filter results by” panel lets you filter complaint data. To clear filters, uncheck individual selections in the filter panel or to select “Clear all filters” under the Search bar.',
    },
    {
      element: GEO.STEP_5.selector,
      intro:
        'You can adjust the date range of the complaint data showed using the “Date CFPB received the complaint” filter. <br /> The default date range is three years from today’s date.',
    },
    {
      element: GEO.STEP_6.selector,
      intro:
        'You can filter complaints by the product and issue the consumer selected when they submitted their complaint.' +
        '<br /><br />' +
        'Expand a product or issue to see its sub-products or sub-issues, where available. ',
    },
    {
      element: GEO.STEP_7.selector,
      intro:
        'The blue show/hide icon allows you to expand or collapse specific filters.',
    },
    {
      element: GEO.STEP_8.selector,
      intro:
        'Use the “Search within” bar to find specific words or phrases in complaints. By default, this will search “All data”, but you can change this using the gray dropdown menu.' +
        '<br /> <br />' +
        'Click “Show advanced search tips” to see some advanced ways to refine your word search.',
    },
    {
      element: GEO.STEP_9.selector,
      intro:
        'As you apply filters and search terms, this will display how many complaints are included out of the total number of complaints published in the database.',
    },
    {
      element: GEO.STEP_10.selector,
      intro:
        'You can export full or filtered results to CSV or JSON by selecting “Export data”.',
    },
    {
      element: GEO.STEP_11.selector,
      intro: 'Select “Print” to generate a pdf of the current page.',
    },
    {
      element: GEO.STEP_12.selector,
      intro:
        'Map shading defaults to total complaints, but you can change this to show complaints per 1,000 population in that state.',
    },
    {
      element: GEO.STEP_13.selector,
      intro:
        'Hover over a tile on the map to view complaint information specific to that state. You can filter to see data specific to a state by selecting the state or using the filter panel.',
    },
    {
      element: GEO.STEP_14.selector,
      intro:
        'View complaint volume by product in the bar chart. Use the blue arrow by each product to view breakdowns by sub-products.' +
        '<br /><br />Once the product is expanded, select the “Visualize” link as shown below, to view trends on sub-products and issues:',
    },
    {
      element: GEO.STEP_15.selector,
      intro:
        'To learn how to use Trends and List views, select the view and then start the tour.',
    },
  ],
  [MODE_LIST]: [
    {
      element: LIST.STEP_1.selector,
      intro:
        '<div>' +
        '<div class="left">' +
        'Welcome to the Consumer Complaint Database!' +
        '<br /><br />' +
        'Complaints the CFPB sends to companies for response are published in the Consumer Complaint Database after the company responds, confirming a commercial relationship with the consumer, or after 15 days, whichever comes first.' +
        '<br /><br />' +
        'Complaints the CFPB refers to other regulators, such as complaints about depository institutions with less than $10 billion in assets, are not published in the database. This database is not a statistical sample of consumers’ experiences in the marketplace.' +
        '</div>' +
        '<div class="right">' +
        '<h4>Index</h4>' +
        '<ul>' +
        listIndex +
        '</ul></div>' +
        '</div>',
      tooltipClass: 'wide',
    },
    {
      element: LIST.STEP_2.selector,
      intro: 'These links provide more information about this database.',
    },
    {
      element: LIST.STEP_3.selector,
      intro:
        'You can change how to view complaint data by selecting the three views we have available — Trends, List, and Map.' +
        '<br /><br />' +
        'You are currently on the List view, which shows individual complaints.' +
        '<br /><br />' +
        'To visualize complaint data with product and issue breakdowns, select Trends view.' +
        '<br /><br />' +
        'To view complaints by state, select Map view.',
    },
    {
      element: LIST.STEP_4.selector,
      intro:
        'The “Filter results by” panel lets you filter complaint data. To clear filters, uncheck individual selections in the filter panel or to select “Clear all filters” under the Search bar.',
    },
    {
      element: LIST.STEP_5.selector,
      intro:
        'You can adjust the date range of the complaint data showed using the “Date CFPB received the complaint” filter. <br /> The default date range is three years from today’s date.',
    },
    {
      element: LIST.STEP_6.selector,
      intro:
        'You can filter complaints by the product and issue the consumer selected when they submitted their complaint.' +
        '<br /><br />' +
        'Expand a product or issue to see its sub-products or sub-issues, where available. ',
    },
    {
      element: LIST.STEP_7.selector,
      intro:
        'The blue show/hide icon allows you to expand or collapse specific filters.',
    },
    {
      element: LIST.STEP_8.selector,
      intro:
        'Use the “Search within” bar to find specific words or phrases in complaints. By default, this will search “All data”, but you can change this using the gray dropdown menu.' +
        '<br /> <br />' +
        'Click “Show advanced search tips” to see some advanced ways to refine your word search.',
    },
    {
      element: LIST.STEP_9.selector,
      intro:
        'As you apply filters and search terms, this will display how many complaints are included out of the total number of complaints published in the database.',
    },
    {
      element: LIST.STEP_10.selector,
      intro:
        'You can export full or filtered results to CSV or JSON by selecting “Export data”.',
    },
    {
      element: LIST.STEP_11.selector,
      intro: 'Select “Print” to generate a pdf of the current page.',
    },
    {
      element: LIST.STEP_12.selector,
      intro: 'Select the number of complaints to view per page here.',
    },
    {
      element: LIST.STEP_13.selector,
      intro:
        'Sort complaints in the list view by “Newest to oldest,” “Oldest to newest,” “Relevance,” and “Relevance (asc).”' +
        '<br /><br />' +
        'Relevancy are based on any applied search terms.',
    },
    {
      element: LIST.STEP_14.selector,
      intro:
        'You can see all complaints or focus on those with consumer complaint narratives.' +
        '<br /><br />' +
        'We publish the consumer’s narrative description of what happened from their complaint if the consumer opts to share it publicly and after taking steps to remove personal information.',
    },
    {
      element: LIST.STEP_15.selector,
      intro:
        'These are your search results. As you apply filters and search terms, your results will display here. This view shows a preview of the complaint.',
    },
    {
      element: LIST.STEP_16.selector,
      intro:
        'Click the blue ID number to see more details about a particular complaint.',
    },
    {
      element: LIST.STEP_17.selector,
      intro:
        'To learn how to use Map and Trend views, select the view and then start the tour.',
    },
  ],
  [MODE_TRENDS]: [
    {
      element: TRENDS.STEP_1.selector,
      intro:
        '<div>' +
        '<div class="left">' +
        'Welcome to the Consumer Complaint Database!' +
        '<br /><br />' +
        'Complaints the CFPB sends to companies for response are published in the Consumer Complaint Database after the company responds, confirming a commercial relationship with the consumer, or after 15 days, whichever comes first.' +
        '<br /><br />' +
        'Complaints the CFPB refers to other regulators, such as complaints about depository institutions with less than $10 billion in assets, are not published in the database. This database is not a statistical sample of consumers’ experiences in the marketplace.' +
        '</div>' +
        '<div class="right">' +
        '<h4>Index</h4>' +
        '<ul>' +
        trendsIndex +
        '</ul></div>' +
        '</div>',
      tooltipClass: 'wide first-step',
    },
    {
      element: TRENDS.STEP_2.selector,
      intro: 'These links provide more information about this database.',
    },
    {
      element: TRENDS.STEP_3.selector,
      intro:
        'You can change how to view complaint data by selecting the three views we have available—Trends, List, and Map.' +
        '<br /><br />' +
        'You are currently on the Trends view, which visualizes complaint data using a trend line with product and issue breakdowns.' +
        '<br /><br />' +
        'To read individual complaints, select List view.' +
        '<br /><br />' +
        'To view complaints by state, select Map view.',
      position: 'top',
    },
    {
      element: TRENDS.STEP_4.selector,
      intro:
        'The “Filter results by” panel lets you filter complaint data. To clear filters, uncheck individual selections in the filter panel or to select “Clear all filters” under the Search bar.',
    },
    {
      element: TRENDS.STEP_5.selector,
      intro:
        'You can adjust the date range of the complaint data showed using the “Date CFPB received the complaint” filter. <br /> The default date range is three years from today’s date.',
    },
    {
      element: TRENDS.STEP_6.selector,
      intro:
        'You can filter complaints by the product and issue the consumer selected when they submitted their complaint.' +
        '<br /><br />' +
        'Expand a product or issue to see its sub-products or sub-issues, where available. ',
    },
    {
      element: TRENDS.STEP_7.selector,
      intro:
        'The blue show/hide icon allows you to expand or collapse specific filters.',
    },
    {
      element: TRENDS.STEP_8.selector,
      intro:
        'Use the “Search within” bar to find specific words or phrases in complaints. By default, this will search “All data”, but you can change this using the gray dropdown menu.' +
        '<br /> <br />' +
        'Click “Show advanced search tips” to see some advanced ways to refine your word search.',
    },
    {
      element: TRENDS.STEP_9.selector,
      intro:
        'As you apply filters and search terms, this will display how many complaints are included out of the total number of complaints published in the database.',
    },
    {
      element: TRENDS.STEP_10.selector,
      intro:
        'You can export full or filtered results to CSV or JSON by selecting “Export data”.',
    },
    {
      element: TRENDS.STEP_11.selector,
      intro: 'Select “Print” to generate a pdf of the current page.',
    },
    {
      element: TRENDS.STEP_12.selector,
      intro:
        'Select the dropdown to aggregate the data by product or by the company to which the CFPB sent the complaint for response.',
    },
    {
      element: TRENDS.STEP_13.selector,
      intro:
        'Select this dropdown to change the time interval displayed on the trend line.',
    },
    {
      element: TRENDS.STEP_14.selector,
      intro:
        'View complaint volume by product in the bar chart. Use the blue arrow by each product to view breakdowns by sub-products.' +
        '<br /><br />Once the product is expanded, select the “Visualize” link as shown below, to view trends on sub-products and issues:',
    },
    {
      element: TRENDS.STEP_15.selector,
      intro:
        'To learn how to use List and Map views, select the view and then start the tour.',
      position: 'top',
    },
  ],
};
