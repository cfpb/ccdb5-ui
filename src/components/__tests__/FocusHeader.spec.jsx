import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import ReduxFocusHeader, {
  FocusHeader,
  mapDispatchToProps,
  mapStateToProps
} from '../Trends/FocusHeader'
import React from 'react'
import renderer from 'react-test-renderer'
import { REQUERY_ALWAYS } from '../../constants'
import thunk from 'redux-thunk'
import { shallow } from 'enzyme'

function setupSnapshot( { focus, results } ) {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    query: {
      focus,
      lens: 'Product',
      subLens: 'sub_product'
    },
    trends: {
      total: 90120,
      results
    }
  } )

  return renderer.create(
    <Provider store={ store }>
      <IntlProvider locale="en">
        <ReduxFocusHeader/>
      </IntlProvider>
    </Provider>
  )
}

describe( 'component:FocusHeader', () => {
  it( 'renders both tabs without crashing', () => {
    const target = setupSnapshot( { focus: '', results: {} } )
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )

  describe( 'buttons', () => {
    let cb = null
    let target = null

    beforeEach( () => {
      cb = jest.fn()
      target = shallow( <FocusHeader clearFocus={ cb }
                                     focus={ 'Focus item' }
                                     lens={ 'Foo' }
                                     total={ '9,123' }
      /> )
    } )

    it( 'changeFocus is called when the button is clicked', () => {
      const prev = target.find( '#clear-focus' )
      prev.simulate( 'click' )
      expect( cb ).toHaveBeenCalled()
    } )

  } )

  describe( 'mapDispatchToProps', () => {
    it( 'hooks into removeFocus', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch ).clearFocus()
      expect( dispatch.mock.calls ).toEqual( [
        [ {
          requery: REQUERY_ALWAYS,
          type: 'FOCUS_REMOVED'
        } ]
      ] )
    } )
  } )

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        query: {
          focus: 'Foo',
          lens: 'Bar'
        },
        trends: {
          total: 1000
        }
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( {
        focus: 'Foo',
        lens: 'Bar',
        total: '1,000'
      } )
    } )
  } )

} )
