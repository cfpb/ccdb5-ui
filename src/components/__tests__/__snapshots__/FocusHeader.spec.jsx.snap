// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:FocusHeader renders both tabs without crashing 1`] = `
<div
  className="focus-header"
>
  <button
    className="a-btn a-btn__link clear-focus"
    id="clear-focus"
    onClick={[Function]}
  >
    <svg
      className="cf-icon-svg"
      viewBox="0 0 559.6 1200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M494.5 1090.7c-17.3 0-33.8-6.8-46-19L19 642.1c-25.4-25.4-25.4-66.5 0-91.9l429.5-429.5c25.6-25.1 66.8-24.8 91.9.8 24.8 25.3 24.8 65.8 0 91.1L156.9 596.2l383.6 383.6c25.4 25.4 25.4 66.5.1 91.9-12.3 12.2-28.8 19-46.1 19z"
      />
    </svg>
    View product trends
  </button>
  <div>
    <section
      className="focus"
    >
      <h1>
        Foo Bar
      </h1>
      <span
        className="divider"
      />
      <h2>
        90,120
         Complaints
      </h2>
    </section>
  </div>
  <div
    className="tabbed-navigation lens"
  >
    <section>
      <button
        className="tab sub_product active"
        onClick={[Function]}
      >
        Sub-products
      </button>
      <button
        className="tab issue"
        onClick={[Function]}
      >
        Issues
      </button>
    </section>
  </div>
</div>
`;
